### PR TITLE
[BugFix][Worker] Fix padded sequence length handling for ViT encoder ACL graph

### PR DESCRIPTION
**Pull request by zhoux77899**
### What this PR does / why we need it?

This PR unifies FIA `actual_seq_lengths` handling for ViT encoder ACL graph across eager, capture, and replay. Previously, `AscendMMEncoderAttention` and `encoder_acl_graph` built sequence lengths independently, which could produce incorrect results for padded `cu_seqlens` and budget-padding trailing zeros.

Fix #10549

Key changes:
- Introduce shared helper `maybe_compute_actual_seq_lengths()` in `encoder_acl_graph.py` to convert `cu_seqlens` → host `list[int]`, filter invalid endpoints, and ensure `actual_seq_lengths[-1] == num_query_tokens`.
- Modified `_maybe_compute_cu_seqlens()` in `AscendMMEncoderAttention` for `None` fallback and CPU normalization.
- Refactor eager/capture/replay paths to use the shared logic; simplify replay context to only `cu_seqlens_cpu`.
- Remove redundant per-layer metadata (`vit_layer_idx`, `fullatt_block_indexes`, `sequence_lengths`, `cu_window_seqlens`) and use `self.scale` instead of `scale_value`.

```mermaid
flowchart TB
    direction TB
    Before["Before: duplicated paths"] --> E1[cu_seqlens / sequence_lengths / cu_window_seqlens]
    E1 --> E2[AscendMMEncoderAttention<br/>local convert + per-layer metadata]
    E2 --> E3[_pad_actual_seq_lengths_for_fia]
    Before --> R1[cu_seqlens_cpu / cu_window_seqlens_cpu<br/>sequence_lengths_cpu + fullatt_block_indexes]
    R1 --> R2[encoder_acl_graph<br/>_maybe_compute_actual_seq_lengths]
    R2 --> R3[_pad_actual_seq_lengths_for_fia]
    E3 --> FIA1[FIA]
    R3 --> FIA1
    After["After: shared pipeline"] --> IN[cu_seqlens]
    IN --> BUILD[maybe_compute_actual_seq_lengths<br/>filter + budget align]
    BUILD --> EAGER[eager / capture]
    BUILD --> REPLAY[replay via cu_seqlens_cpu context]
    EAGER --> FIA2[FIA actual_seq_lengths]
    REPLAY --> FIA2
```

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

- The following UT added.
  - `tests/ut/worker/test_encoder_acl_graph.py`
  - `tests/ut/ops/test_mm_encoder_attention.py`
- Also test on `Qwen/Qwen3.6-27B`

```diff
| dataset | version | metric | mode | vllm-api-stream-chat |
|----- | ----- | ----- | ----- | -----|
- | videomme | 2f2602 | [Overall][overall] | gen | 67.78 |
+ | videomme | 2f2602 | [Overall][overall] | gen | 68.85 |
```

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/1f486d96a17303ce8db8e02be39545b2be338446

### CODE REVIEW

File: vllm_ascend/ops/mm_encoder_attention.py [hunk 1]
```diff
@@ -241,11 +249,12 @@ def _forward_eager_fia(
         bsz: int,
         q_len: int,
     ) -> torch.Tensor:
-        actual_seq_lengths_q, actual_seq_lengths_kv = self._maybe_compute_actual_seq_lengths(  # TODO
-            bsz,
-            q_len,
-            cu_seqlens,
-            seq_lens,
+        actual_seq_lengths_q, actual_seq_lengths_kv = self._maybe_compute_actual_seq_lengths(
+            num_query_tokens=query.shape[0],
```
Comment (winson-00178005) [inline vllm_ascend/ops/mm_encoder_attention.py:206]: ## 重构质量良好，一处建议

整体重构方向正确：将 `actual_seq_lengths` 的构建逻辑统一提取到 `build_fia_actual_seq_lengths`，并在 `encoder_acl_graph.py` 中增加 `_align_fia_endpoints_to_num_tokens` 处理边界情况。单元测试覆盖充分。

一处建议：`_maybe_compute_actual_seq_lengths` 的参数名从 `seq_lens` 改为 `sequence_lengths`，但调用端（`_forward_eager_fia`、`_forward_capture_fia`）传入的变量名仍然叫 `seq_lens`。虽然 keyword-only 传参不影响功能，但局部变量名和关键字参数名不一致可能造成阅读混淆。建议统一命名。

Comment (zhoux77899) [reply vllm_ascend/ops/mm_encoder_attention.py:206]: Following upstream naming conventions, the cumulative sum variable `cumulative_sequence_lengths` is renamed to `cu_seqlens`, while the others remain as `sequence_lengths`.
The signatures of `forward_oot`, `_forward_eager_fia`, and `_forward_capture_fia` have been unified to use the upstream of `cu_seqlens` and `sequence_lengths`.

File: vllm_ascend/worker/encoder_acl_graph.py [hunk 2]
```diff
@@ -135,15 +136,96 @@ def set_encoder_forward_context(
...
+        if not filtered or end > filtered[-1]:
+            filtered.append(end)
+
+    # 4. Ensure terminal endpoint equals num_tokens.
+    if not filtered or filtered[-1] != num_tokens:
+        filtered.append(num_tokens)
+    return filtered
+
+
+def _cumulative_buffer_to_fia_endpoints(buffer: torch.Tensor) -> list[int]:
```
Comment (weijinqian0) [inline vllm_ascend/worker/encoder_acl_graph.py]: plz move this to the place it is used.

File: vllm_ascend/worker/encoder_acl_graph.py [hunk 3]
```diff
@@ -135,15 +136,96 @@ def set_encoder_forward_context(
...
+
+    Input is ``cu_seqlens`` / ``cu_window_seqlens`` style: leading 0 is a format
+    marker, not a sequence boundary. Output may contain trailing padding zeros and
+    need not end at the token budget — callers run ``_align_fia_endpoints_to_num_tokens``.
+    """
+    flat = buffer.detach().cpu().view(-1).tolist()
+    return flat[1:] if flat else flat
+
+
+def _per_sequence_lengths_to_fia_endpoints(buffer: torch.Tensor) -> list[int]:
```
Comment (weijinqian0) [inline vllm_ascend/worker/encoder_acl_graph.py]: same to last one.

**Review by yiz-liu — APPROVED** [review]

**Comment by yiz-liu** [comment]
/cherry-pick releases/v0.23.0
[Bot]: cherry-pick completed successfully. New PR created: https://github.com/vllm-project/vllm-ascend/pull/11943

**Comment by yiz-liu** [comment]
/revert
[Bot]: revert completed successfully. New PR created: https://github.com/vllm-project/vllm-ascend/pull/11994
